### PR TITLE
feat(client,server): database deletion functionality

### PR DESCRIPTION
**Pull request by Akirtovskis**
Can be merged after : https://github.com/ledokku/ledokku/pull/88 . 

To test try creating db and deleting in the database views settings part.

### CODE REVIEW

**Comment by pradel** [comment]
@Akirtovskis conflicts on this one :grin:

**Comment by Akirtovskis** [comment]
@pradel and they are gone! :)

File: server/src/index.ts [hunk 1]
```diff
@@ -208,3 +218,10 @@ http.listen({ port: 4000 }, () =>
     `🚀 Server ready at http://localhost:4000${apolloServer.graphqlPath}`
   )
 );
+
+const test = async () => {
```
Comment (pradel) [inline server/src/index.ts:222]: let's remove this?

File: server/src/graphql/mutations/destroyDatabase.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  { input },
+  { userId }
+) => {
+  if (!userId) {
+    throw new Error('Unauthorized');
+  }
+
+  const { databaseId } = input;
+
+  // TODO check name of the database is unique per server
```
Comment (pradel) [inline server/src/graphql/mutations/destroyDatabase.ts:17]: This TODO can be removed

File: server/src/graphql/mutations/destroyDatabase.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const ssh = await sshConnect();
+
+  // We delete the database
+  await prisma.database.delete({
+    where: {
+      id: databaseId,
+    },
+  });
+
+  const result = await dokku.postgres.destroy(ssh, databaseToDelete.name);
```
Comment (pradel) [inline server/src/graphql/mutations/destroyDatabase.ts:39]: I guess it make more sense to move this one before the prisma call. That way if dokku fail to delete the db, it's not removed from the database :)

File: server/src/index.ts (hunk 1 above)
Comment (Akirtovskis) [reply server/src/index.ts]: 🤦 for leaving this in active PR.

**Pull request merged**